scripts/apps/monitoring/directives/MonitoringFilteringButtons.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable react/no-multi-comp */
+
+import React from 'react';
+import {IMonitoringFilter, IDesk} from 'superdesk-api';
+import {extensions} from 'appConfig';
+import {flattenDeep, get} from 'lodash';
+import {dataApiByEntity} from 'core/helpers/CrudManager';
+import {Badge, Button} from 'superdesk-ui-framework';
+
+interface IProps {
+    deskId: IDesk['_id'];
+    isFilterActive(button: IMonitoringFilter): boolean;
+    toggleFilter(button: IMonitoringFilter): void;
+    addResourceUpdatedEventListener: any;
+
+    // `activeFilters` isn't meant to be read
+    // it's only required so angular re-renders the component when filters change
+    activeFilters: never;
+}
+
+interface IState {
+    matchingItemsCount?: number;
+    currentItems?: {[key: string]: {[key: string]: any}};
+}
+
+interface IPropsFilterButton extends IProps {
+    button: IMonitoringFilter;
+}
+
+class FilterButton extends React.PureComponent<IPropsFilterButton, IState> {
+    getFilters: () => {[key: string]: any};
+
+    constructor(props: IPropsFilterButton) {
+        super(props);
+
+        this.state = {};
+
+        this.getFilters = () => ({'task.desk': [this.props.deskId], ...this.props.button.query});
```

Comment:
why is `getFilters` a function instead of a plain object?

Reply:
because `getFilters` depends on props and is initialized in the constructor. When props change, the constructor is not called again. If it were a plain object it would get stuck with initial values and wouldn't react to prop changes.